### ISSUE TITLE
Output format when indel not found in prescored file

### ISSUE THREAD

**Issue by jmo616** [kircherlab/CADD-scripts#14]
Hello,

I'm using CADDv1.5 (GRCh38) to score a large number of SNVs and indels, and wanted to know how to interpret output and an error message when an indel in my VCF is potentially not found in the prescored file. I can't be sure, but it seems as though CADD returns indels that are upstream/downstream of that position if the ref/alt alleles match what was in my VCF? 

Here are the commands and previews of my files:
% ./CADD.sh -a -g GRCh38 -o CADD_ExampleOutput_withAnno.tsv.gz ExampleInput.vcf.gz 
CADD-v1.5 (c) University of Washington, Hudson-Alpha Institute for Biotechnology and Berlin Institute of Health 2013-2019. All rights reserved.
Opening /shared/CADD/CADD-scripts/data/prescored/GRCh38_v1.5/incl_anno/InDels_inclAnno.tsv.gz...
Opening /shared/CADD/CADD-scripts/data/prescored/GRCh38_v1.5/incl_anno/whole_genome_SNVs_inclAnno.tsv.gz...
Input file /tmp/tmp.sVxd4dNddm/ExampleInput.csv.gz is empty.

CADD scored variants written to file: CADD_ExampleOutput_withAnno.tsv.gz

My input VCF (file attached):
10      104817  .       C       CCA
10      104818  .       C       A
10      104818  .       C       T
10      104820  .       C       CCA

CADD output (file attached w/full anno):
10      104816  C       CCA     INS ...
10      104817  C       CCA     INS ...
10      104818  C       A       SNV ...
10      104818  C       CCA     INS ...
10      104818  C       T       SNV ...
10      104819  C       CCA INS ...

I had thought CADD was matching strictly on chr:pos_ref/alt, but that wouldn't explain the output that I'm getting, since the positions for the indels don't match my input VCF, even when you account for the length of the insertion. This output also seems different than what's produced using the test/input.vcf (in terms of positions being returned that weren't in my input VCF), and the issue also doesn't occur for every indel in my dataset. Since CADD uses Ensembl, I ran the most recent version of Ensembl-VEP (98.3) with my same input VCF and got the following result (file attached). However, I realize that CADDv1.5 is using Ensembl 95, so it's not a direct comparison:

% ./vep --cache --dir_cache .vep/ -i ExampleInput.vcf.gz --show_ref_allele --symbol --per_gene --no_stats -o Ensembl_ExampleOutput

10_104818_-/CA  10:104817-104818
10_104818_C/A   10:104818
10_104818_C/T   10:104818
10_104821_-/CA  10:104820-104821

I also got an odd error message when running CADD with my example input VCF (above), which seems to correspond to a tmp CSV file that's being created (and is subsequently empty) when calling the 'trackTransformation.py' in the 'CADD.sh' script. 

Please let me know if you have any questions, and I greatly appreciate any help in clarifying this issue! Thanks in advance,
Jackie

[CADD_ExampleOutput_withAnno.tsv.gz](https://github.com/kircherlab/CADD-scripts/files/3990224/CADD_ExampleOutput_withAnno.tsv.gz)
[ExampleInput.vcf.gz](https://github.com/kircherlab/CADD-scripts/files/3990225/ExampleInput.vcf.gz)
[Ensembl_ExampleOutput.txt](https://github.com/kircherlab/CADD-scripts/files/3990259/Ensembl_ExampleOutput.txt)

**Comment by makirc** [comment]
Dear Jackie,

Thank you for your message. I am not sure about the error message and we probably need to look at that once back at the office in January, but I wanted to briefly answer the question about changing InDel coordinates. Through Ensembl VEP and our own scripts trimming additional context bases, InDels are reported with the left-most coordinate and with minimal sequence context. Both are standard practice for reporting InDels in VCF. The left-most coordinate reporting is relevant for all simple repeats, where the insertion/deletion of repeat units cannot be assigned. Here, the left most coordinate should be reported. Most variant callers should be implementing it like that, but not all do. In this case, variants may shift. Further, some variant callers (e.g. GATK) are combining overlapping InDel calls together in one VCF line, causing additional context bases (i.e. bases that are not actually changed and reported for the REF and ALT alleles). Since we are separating the effect prediction for each variant, we are trimming these additional context bases and this may also result in a coordinate shift of the variant.

I hope this helps and we will get back to you about the error message after looking at your files in the beginning of next year.

Happy holidays,

Martin

**Comment by jmo616** [comment]
Hi Martin,

Thank you for such a quick response--yes, I think that helps a bit in clarifying the presence of coordinate shifts, and I'll look more into how our indels were called in order to try and resolve the discrepant variants. 

Thanks for looking into the error message when you're back in the office, and happy holidays to you as well!

Best,
Jackie

**Comment by aerval** [comment]
Dear Jackie,

Thank you for discovering and reporting this issue. The problem was that prescored InDel with similar positions and same ref and alt allele were reported by mistake.

In order to update your scripts you only have to update the file `src/scripts/extract_scored.py`. However you can also download the CADD zip archive again and extract it. Now further installation is necessary.

Thank you very much,
Philipp

**Comment by jmo616** [comment]
Hi Philipp,

Thanks so much for looking into this and letting me know--I downloaded the new 'extract_scored.py' and encountered an error due to an extra close parentheses in the code for the position check on line 56 here, so wanted to alert you to that:

if (vfields[fref] == lref) and (vfields[falt] == allele) and (vfields[fpos] == pos)**)**:

Once I deleted the second close parentheses at the end of the line, everything ran fine. So far my output files do not have indels with discrepant positions, so I think it's working properly. Thank you again!

Best,
Jackie
